### PR TITLE
chore: run CI on latest Node.js

### PR DESCRIPTION
**Pull request by ehmicky**
Part of https://github.com/netlify/team-dev/issues/19

This runs the CI on the `latest` Node.js version.

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -83,7 +81,7 @@ jobs:
       - uses: actions/upload-artifact@v2
         if: always()
         with:
-          name: cypress-results-node-${{ matrix.node-version }}-os-${{ matrix.os }}
+          name: cypress-results-node-os-${{ matrix.os }}
```
Comment (erezrokah) [inline .github/workflows/workflow.yml:84]: 👍

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -53,18 +54,15 @@ jobs:
           file: coverage/coverage-final.json
           flags: ${{ steps.test-coverage-flags.outputs.os }},${{ steps.test-coverage-flags.outputs.node }}
   cypress-tests:
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-        node-version: [15]
     runs-on: ${{ matrix.os }}
```
Comment (erezrokah) [inline .github/workflows/workflow.yml:57]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/workflow.yml (hunk 1 above)
Comment (erezrokah) [inline .github/workflows/workflow.yml:84]:
```suggestion
          name: cypress-results
```

Comment (ehmicky) [reply .github/workflows/workflow.yml]: Note: the reason this was removed is because Cypress does not allow `*` in there.

**Review by erezrokah — APPROVED** [review]

**Pull request merged**